# Hugging Face Embedding Evaluation

Stress the agent loop with dataset streaming plus model loading. The default target streams `fancyzhx/ag_news` and embeds rows with `sentence-transformers/all-MiniLM-L6-v2`.

Useful knobs: `HF_EMBED_DATASET`, `HF_EMBED_CONFIG`, `HF_EMBED_SPLIT`, `HF_EMBED_TEXT_FIELD`, `HF_EMBED_LABEL_FIELD`, `HF_EMBED_ROWS`, `HF_EMBED_MODEL`.

In [1]:
from pathlib import Path

exec(Path("samples/hf_embedding_eval.py").read_text(encoding="utf-8"))


dataset: fancyzhx/ag_news
split: train
rows_embedded: 80
model: sentence-transformers/all-MiniLM-L6-v2


Batches: 100%|██████████| 3/3 [00:02<00:00,  1.39it/s]

label_counts:
- 2: 78
- 3: 2
nearest_centroid_self_check_accuracy: 0.650


## Follow-up Work

After the embedding run, inspect model load time, batch throughput, label balance, and whether nearest-centroid accuracy is plausible enough to justify deeper evaluation.

## Agent Finding - Notebook Lens CLI Run

`run-clean --timeout 900` was executed with `HF_EMBED_ROWS=80` and the Hugging Face token loaded from the research-suite environment. The notebook completed successfully in code cell `hfembed1`.

The previous dataset loading failure was resolved by changing the default stream target from unnamespaced `ag_news` to `fancyzhx/ag_news`.

The run embedded 80 rows with `sentence-transformers/all-MiniLM-L6-v2`, emitted label counts, and reported `nearest_centroid_self_check_accuracy: 0.650`.

Notebook Lens now surfaces the relevant stdout/stderr excerpt directly from `run-clean`, while `show-cell --outputs full` can render the detailed cell output without the previous default truncation.
